shared/templates/create_mount_options.py

```diff
@@ -8,83 +8,124 @@
+            self._output_id_template.format(
+                point_id=self._point_id, mount_option=self._mount_option)
+        )
+        self._assert_mount_exists = assert_mount_exists
+        self._process()
 
-    def process(self, mount_point, mount_option, point_id, assert_mount_exists, template_file):
+    def _process(self):
         raise NotImplementedError("You are supposed to use a derived class.")
 
-    def process_with_variable(
-            self, mount_point, mount_option, point_id, assert_mount_exists, template_file):
-        raise NotImplementedError("You are supposed to use a derived class.")
+    def _set_correct_values(self, mount_point, mount_option, point_id):
+        self._mount_point = mount_point
+        self._point_id = point_id
+        self._mount_option = mount_option
 
 
 class RemediationTarget(MountOptionTarget):
-    def process(self, mount_point, mount_option, point_id, assert_mount_exists,
-                template_file, stem=""):
-        if len(stem) == 0:
-            stem = point_id + '_' + mount_option
-        mount_has_to_exist = "yes" if assert_mount_exists else "no"
+    TEMPLATE_FILE_BASE = "./template_{target}_mount_option"
+
+    def _process(self):
+        mount_has_to_exist = "yes" if self._assert_mount_exists else "no"
         self.generator.file_from_template(
-            template_file,
+            self.template_file,
             {
                 "%MOUNT_HAS_TO_EXIST%": mount_has_to_exist,
-                "%MOUNTPOINT%": mount_point,
-                "%MOUNTOPTION%": re.sub(' ', ',', mount_option),
+                "%MOUNTPOINT%": self._mount_point,
+                "%MOUNTOPTION%": re.sub(' ', ',', self._mount_option),
             },
-            self.output_format_string,
-            stem
+            self._output_fname,
+            ""
         )
 
-    def process_with_variable(self, mount_point, mount_option, point_id, assert_mount_exists,
-                              template_file):
-        # e.g. var_removable_partition -> removable_partitions
-        point_id = re.sub(r"^var_(.*)", r"\1s", mount_point)
-        template_file = "{0}_var".format(template_file)
-        stem = "_{0}_{1}".format(mount_option, point_id)
-        return self.process(mount_point, mount_option, point_id,
-                            assert_mount_exists, template_file, stem)
+    def _set_correct_values(self, mount_point, mount_option, point_id):
+        super(RemediationTarget, self)._set_correct_values(mount_point, mount_option, point_id)
+        if mount_point.startswith("var_"):
+            self._point_id = re.sub(r"^var_(.*)", r"\1s", mount_point)
+            self.template_file = "{0}_var".format(self.TEMPLATE_FILE_BASE)
+            self._output_id_template = "{mount_option}_{point_id}"
+        elif not mount_point.startswith("/"):  # no path, but not a variable either
+            raise Skipped("No template available yet")
```

Comment:
I can't inform anybody here, because it would make the build process a UX disaster. So I just left some comments, so it will help when the build system will be refactored.